### ISSUE TITLE
Broken Intel Mac wheels

### ISSUE THREAD

**Issue by FFroehlich** [sys-bio/roadrunner#1177]
After installing libroadrunner 2.5.0 (via `pip install libroadrunner`) on an old Intel Mac (2017 MBP, 2.9 GHz Quad-Core i7, python3.10) importing libroadrunner fails with the following error:

```
>>> import roadrunner
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/ffroehlich/Documents/GitHub/pysb/venv/lib/python3.10/site-packages/roadrunner/__init__.py", line 10, in <module>
    from .roadrunner import *
  File "/Users/ffroehlich/Documents/GitHub/pysb/venv/lib/python3.10/site-packages/roadrunner/roadrunner.py", line 15, in <module>
    from . import _roadrunner
ImportError: dlopen(/Users/ffroehlich/Documents/GitHub/pysb/venv/lib/python3.10/site-packages/roadrunner/_roadrunner.so, 0x0002): tried: '/Users/ffroehlich/Documents/GitHub/pysb/venv/lib/python3.10/site-packages/roadrunner/_roadrunner.so' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64h' or 'x86_64')), '/System/Volumes/Preboot/Cryptexes/OS/Users/ffroehlich/Documents/GitHub/pysb/venv/lib/python3.10/site-packages/roadrunner/_roadrunner.so' (no such file), '/Users/ffroehlich/Documents/GitHub/pysb/venv/lib/python3.10/site-packages/roadrunner/_roadrunner.so' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64h' or 'x86_64'))
```

I believe a related issues led to segfaults when instantiating RoadRunner instances on github actions runners (see https://github.com/pysb/pysb/actions/runs/7644052735/job/20827461125, python 3.8-3.11).

We were able to resolve both issues by pinning libroadrunner to 2.4.0. I was not able to reproduce the issue on newer arm macs.

**Comment by adelhpour** [comment]
Could you please tell us which version of macOS you are using in your old Intel Mac?

**Comment by FFroehlich** [comment]
it's macOS13. I did some further digging and found that I can reproduce the segfault locally using the `12_0_universal2` wheel, the `13_0_universal2` wheel produces the incompatible architecture and the `11_0_x86_64` wheel actually works fine.

**Comment by adelhpour** [comment]
We believe that might be the problem with the universal binaries, which may have some arm64 or x86_64 dependencies. We are working on it to find the best solution to fix this issue! We will get back to you with an update once we made our final decision. Thanks for the checking this out.

**Comment by luciansmith** [comment]
OK, I've pulled all the wheel files named 'universal' because they were not, in fact, universal.  This should hopefully leave the files that will actually work on your platform to be the ones pip grabs for your system.  If this isn't true, let us know!

**Comment by FFroehlich** [comment]
Thanks, Looks like everything is working fine now!

One thing that I noticed though is that the mac `_x86_64` wheel for python 3.8 also, despite it's name, doesn't appear to be a `_x86_64` wheel (see https://github.com/pysb/pysb/actions/runs/7655573514/job/20861971848). The latest version that is available on intel macs running python 3.8 appears to be 2.2.2 (see https://github.com/pysb/pysb/actions/runs/7671183673/job/20909010927).

**Comment by luciansmith** [comment]
That's truly odd; that file was automatically built on azure along with every single other x86 wheel file.  Maybe there's an issue with the particular version of 3.8?  

I was planning to drop 3.8 now anyway, moving up to 3.12, but it's worth keeping an eye on it.  Thanks for testing it!
